Added YouTube Activity Video & Improved Activity Selection

elsa-bismuth opened the pull request:
# Added YouTube Activity Video & Improved Activity Selection

## :recycle: Current situation & Problem
This PR addresses the need to enhance user experience when selecting activities in the add sheet view. The previous picker was not optimized for quick selection, making it less intuitive. Additionally, a YouTube link for an activity video has been added to provide users with better guidance.

## :gear: Release Notes 
Added a link to a YouTube video for activity guidance.
Updated the activity selection picker in the add sheet view to make it easier for users to select one of six activities.

<img width="392" alt="Capture d’écran 2025-03-12 à 15 39 33" src="https://github.com/user-attachments/assets/9a8cf0a4-b86e-4a8d-8935-5ab4cfcdc7e4" />

<img width="373" alt="Capture d’écran 2025-03-12 à 15 40 37" src="https://github.com/user-attachments/assets/1503b254-066b-473d-9044-89556fa49e3b" />

<img width="383" alt="Capture d’écran 2025-03-12 à 15 39 20" src="https://github.com/user-attachments/assets/cd8c5ee1-52fe-4a3a-a297-39c70b6e4b20" />
d-a9a2-67fc14492c9e" />

## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 


## :white_check_mark: Testing
TBD

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

elsa-bismuth committed:
add video link to activity recommendations

elsa-bismuth committed:
add activity cards button to activity sheet view

kkellybonilla reviewed (changes requested):
i think you should change the hyperlink they’re not necessarily sport videos, you can change it to be “Click HERE for videos with activity ideas”

& the buttons shouldn’t be on the sheet, it should be on the tab view itself

elsa-bismuth committed:
update activity add UI

elsa-bismuth committed:
corrected periphery errors

elsa-bismuth committed:
Merge branch 'main' into elsa-bismuth/update-activity-ui

elsa-bismuth committed:
added streak to display milestone message call

for some reason the pushed code wasnt matching with what I have on VSCode (i still cant use Xcode)

elsa-bismuth committed:
updated 5min increments to 10min increments

elsa-bismuth commented on Stanford360/Activity/View/ActivityComponentsView.swift:
Created this new file with reusable UI components

elsa-bismuth commented on Stanford360/Activity/View/ActivityComponentsView.swift:
```diff
@@ -0,0 +1,308 @@
+    @Binding var selectedDate: Date
+    var title: String = "When did you do it?"
+    var dateRange: PartialRangeThrough<Date> = ...Date() // Default to past dates
+    var displayComponents: DatePickerComponents = [.date]
```
only shows day, not time anymore

elsa-bismuth commented on Stanford360/Activity/View/ActivityComponentsView.swift:
```diff
@@ -0,0 +1,308 @@
+}
+
+// MARK: - MinutesInputView
+struct MinutesInputView: View {
```
this creates a slider to choose the number of minutes in increments of 10min

elsa-bismuth committed:
removed the errorDisplay() function in the Activity components

it still exists in hydration module

elsa-bismuth committed:
Update ActivityAddView.swift alignment

elsa-bismuth committed:
Update ActivityAddView.swift

updated streak to conform to new displayMilestoneMessage

elsa-bismuth committed:
Merge branch 'main' into elsa-bismuth/update-activity-ui
